cumulusci/tasks/github_commit.py

```diff
@@ -0,0 +1,217 @@
+import base64
+import hashlib
+import io
+import logging
+import os
+
+from cumulusci.core.exceptions import GithubException
+
+
+class CommitDir(object):
+    """Commit all changes in local_dir to branch/repo_dir"""
+
+    def __init__(self, repo, logger=None, author=None):
+        """
+        repo: github3.py Repository object, authenticated
+        logger (optional): python logger object
+        author (optional): dict of name, email, time - default uses auth'd info
+        """
+        self.repo = repo
+        if not logger:
+            logger = logging.getLogger(os.path.basename(__file__))
+        self.logger = logger
+        self.author = author if author else {}
+
+    def __call__(self,
+            local_dir,
+            branch,
+            repo_dir=None,
+            commit_message=None,
+            dry_run=True,
+        ):
+        """
+        local_dir: path to local directory to commit
+        branch: target branch name
+        repo_dir: target path within repo - use '' for repo root
+        commit_message: message for git commit
+        dry_run: skip creating GitHub data if True
+        """
+
+        # prepare dir args
+        local_dir = os.path.abspath(local_dir)
+        if not os.path.isdir(local_dir):
+            raise GithubException('Not a dir: {}'.format(local_dir))
+        if repo_dir is None:
+            repo_dir = ''
+        if repo_dir.startswith('.'):
+            repo_dir = repo_dir[1:]
+        if repo_dir.startswith('/'):
+            repo_dir = repo_dir[1:]
+        if repo_dir.endswith('/'):
+            repo_dir = repo_dir[:-1]
```

Comment:
We don't want to use anything from os.path because this is not dealing with local paths. Also we want to preserve any "inner" dir seps e.g. `./foo/bar/` should be converted to `foo/bar`.